en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -6,190 +6,154 @@ iconBrand: php
+
+## Class Member Ordering
 
-```php
-MyClass::get()->where(['"Score" > ?' => 50]);
+Order code in classes in the following order:
 
-```
+- Class constants
+- Data-model definition static properties (`$db`, `$has_one`, `$many_many`, etc)
+- Static properties
+- Member properties
+- Static methods
+- Controller action methods
+- Commonly used methods like `getCMSFields()`
+- Accessor methods (`getMyField()` and `setMyField()`)
+- Template data-access methods (methods that will be called by a `$MethodName` or `<% loop $MethodName %>` construct in a template somewhere)
+- Object methods
 
-It is preferable to use parameterised queries whenever necessary to provide conditions
-to a SQL query, where values placeholders are each replaced with a single unquoted question mark.
-If it's absolutely necessary to use literal values in a query make sure that values
-are single quoted.
+## Inline comments
 
+- If there is a non-obvious reason for a code operation, include inline comments that explain *why* we are doing this operation. Remember that while something may be obvious to you now, it will not be obvious to someone else in 6 months.
+- Self explanatory code with well-named methods and variables is preferred to inline comments that explain *what* the code does, though it's still OK to include these sorts of comments if you think it makes the code easier to follow. This is often the case for larger methods, though it may be better to simply split those methods into several private methods.
 
-```php
-MyClass::get()->where("\"Title\" = 'my title'");
-```
+### Unit testing
 
-Use [ANSI SQL](https://en.wikipedia.org/wiki/SQL#Standardization) format where possible.
+- Unit test method names should match the method that they're testing with the first letter of the method uppercased and the word `test` prefixed (e.g. for the method `myMethodName()` the unit test method should be called `testMyMethodName()`).
+- Aim to use the `@dataProvider` annotation to provide test case data as it makes code much cleaner and it makes it very easy to add additional test cases.
+- Data provider method names should be same as the test case method name they're providing for with the leading work `test` substituted for `provide` (e.g. for `testSomething()` the DataProvider is `provideSomething()`).
+- Data provider array keys should describe the scenario they're testing.
+- Unless you need to explicitly create dynamic fixtures, fixtures should be added via yaml fixture files.
+- It's usually OK to use reflection to change the visibility of a private or protected method so that it can be tested independently. It is generally preferable to keep a smaller public API surface and use reflection to test functionality, rather than making the API public just so it can be tested.
 
-## Secure Development
+## Raw SQL
 
-See [security](/developer_guides/security) for conventions related to handing security permissions.
```

Comment:
Add the secure development section and link back in.

Reply:
It's not related to code conventions at all?

It just seems like a random link

Reply:
Discussed offline - agreed that this page is essentially for things that can be linted, sniffed, or detected through fairly basic static analysis and therefore the security guidelines are out of scope.